sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/batch/BulkExecutorTest.java

```diff
@@ -0,0 +1,354 @@
+import com.azure.cosmos.models.PartitionKey;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.SignalType;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BulkExecutorTest extends BatchTestBase {
+
+    private CosmosAsyncClient client;
+    private CosmosAsyncContainer container;
+    private CosmosAsyncDatabase database;
+    private String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
+
+    @Factory(dataProvider = "clientBuilders")
+    public BulkExecutorTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @AfterClass(groups = { "emulator" }, timeOut = 3 * SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        logger.info("starting ....");
+        safeDeleteDatabase(database);
+        safeCloseClient(client);
+    }
+
+    @AfterMethod(groups = { "emulator" })
+    public void afterTest() throws Exception {
+        if (this.container != null) {
+            try {
+                this.container.delete().block();
+            } catch (CosmosException error) {
+                if (error.getStatusCode() != 404) {
+                    throw error;
+                }
+            }
+        }
+    }
+
+    @BeforeMethod(groups = { "emulator" })
+    public void beforeTest() throws Exception {
+        this.container = null;
+    }
+
+    @BeforeClass(groups = { "emulator" }, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosContainerTest() {
+        client = getClientBuilder().buildAsyncClient();
+        database = createDatabase(client, preExistingDatabaseId);
+    }
+
+    static protected CosmosAsyncContainer createContainer(CosmosAsyncDatabase database) {
+        String collectionName = UUID.randomUUID().toString();
+        CosmosContainerProperties containerProperties = getCollectionDefinition(collectionName);
+
+        database.createContainer(containerProperties).block();
+        return database.getContainer(collectionName);
+    }
+
+    static protected CosmosAsyncDatabase createDatabase(CosmosAsyncClient client, String databaseId) {
+        CosmosDatabaseProperties databaseSettings = new CosmosDatabaseProperties(databaseId);
+        client.createDatabase(databaseSettings).block();
+        return client.getDatabase(databaseSettings.getId());
+    }
+
+    static protected CosmosAsyncDatabase createDatabaseIfNotExists(CosmosAsyncClient client, String databaseId) {
```

Comment:
we have method in cosmosClient.createDatabaseIfNotExists -> why not directly use it?

Reply:
Fixed